pkg/audit/manager_test.go

```diff
@@ -39,3 +94,240 @@ func Test_mergeErrors(t *testing.T) {
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "v1",
+						"kind":       "Namespace",
+						"metadata": map[string]interface{}{
+							"name": "test-namespace-1",
+						},
+					},
+				},
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "v1",
+						"kind":       "Namespace",
+						"metadata": map[string]interface{}{
+							"name": "test-namespace-2",
+						},
+					},
+				},
+			},
+			want: map[string]*corev1.Namespace{
+				"test-namespace-1": {
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Namespace",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-namespace-1",
+					},
+				},
+				"test-namespace-2": {
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Namespace",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-namespace-2",
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := nsMapFromObjs(tt.objs)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("nsMapFromObjs() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("nsMapFromObjs() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getViolationRef(t *testing.T) {
+	type args struct {
+		gkNamespace string
+		rkind       string
+		rname       string
+		rnamespace  string
+		ckind       string
+		cname       string
+		cnamespace  string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *corev1.ObjectReference
+	}{
+		{
+			name: "Test case 1",
+			args: args{
+				gkNamespace: "default",
+				rkind:       "Pod",
+				rname:       "my-pod",
+				rnamespace:  "default",
+				ckind:       "LimitRange",
+				cname:       "my-limit-range",
+				cnamespace:  "default",
+			},
+			want: &corev1.ObjectReference{
+				Kind:      "Pod",
+				Name:      "my-pod",
+				UID:       "Pod/default/my-pod/LimitRange/default/my-limit-range",
+				Namespace: "default",
+			},
+		},
+		{
+			name: "Test case 2",
+			args: args{
+				gkNamespace: "kube-system",
+				rkind:       "Service",
+				rname:       "my-service",
+				rnamespace:  "default",
+				ckind:       "PodSecurityPolicy",
+				cname:       "my-pod-security-policy",
+				cnamespace:  "kube-system",
+			},
+			want: &corev1.ObjectReference{
+				Kind:      "Service",
+				Name:      "my-service",
+				UID:       "Service/default/my-service/PodSecurityPolicy/kube-system/my-pod-security-policy",
+				Namespace: "kube-system",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getViolationRef(tt.args.gkNamespace, tt.args.rkind, tt.args.rname, tt.args.rnamespace, tt.args.ckind, tt.args.cname, tt.args.cnamespace); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getViolationRef() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getFilesFromDir(t *testing.T) {
```

Comment:
We probably want to test the round-tripping of writing/reading list results in order to catch inconsistencies (since the reading/writing logic is necessarily coupled).

Doing this would probably require some refactoring, which I could see being part of a larger effort later.

Reply:
are we okay to do this as a follow-up?